### PR TITLE
Check iss in odic sign-out

### PR DESCRIPTION
**Pull request by Kahbazi**
#4676
I added the same check that `sid` have for `iss`.

As stated in here https://openid.net/specs/openid-connect-frontchannel-1_0.html#ExampleFrontchannel OpenId Providers can decide whether to send `sid` and `iss` or not. So do we need to have this option in `OpenIdConnectOptions`?

### CODE REVIEW

**Comment by HaoK** [comment]
Can we add some tests for this?

**Comment by Kahbazi** [comment]
We add tests for Missing Issuer and Invalid Issuer but I couldn't think of any way to Assert that those scenarios   actually occur. Since without assertion those tests are not useful, should I keep them?

File: src/Security/Authentication/test/OpenIdConnect/OpenIdConnectTests.cs [hunk 1]
```diff
@@ -294,6 +295,50 @@ public async Task SignOut_WithMissingConfig_Throws()
...
+            {
+                var claimsIdentity = new ClaimsIdentity("Cookies");
+                claimsIdentity.AddClaim(new Claim("iss", "test"));
+                await context.SignInAsync(new ClaimsPrincipal(claimsIdentity));
+            });
+
+            var signInTransaction = await server.SendAsync(DefaultHost);
+
+            var remoteSignOutTransaction = await server.SendAsync(DefaultHost + "/signout-oidc", signInTransaction.AuthenticationCookieValue);
+            Assert.Equal(HttpStatusCode.OK, remoteSignOutTransaction.Response.StatusCode);
```
Comment (Tratcher) [inline src/Security/Authentication/test/OpenIdConnect/OpenIdConnectTests.cs]: You can assert the absence of the Set-Cookie header that was skipped.
https://github.com/aspnet/AspNetCore/pull/6378/files#diff-77dbc837ce88da9bcf6a630ae8969190R164

File: src/Security/Authentication/test/OpenIdConnect/OpenIdConnectTests.cs [hunk 2]
```diff
@@ -294,6 +295,50 @@ public async Task SignOut_WithMissingConfig_Throws()
             Assert.Equal("Cannot redirect to the end session endpoint, the configuration may be missing or invalid.", exception.Message);
         }
 
+        [Fact]
+        public async Task SignOut_WithMissingIssuer()
```
Comment (Tratcher) [inline src/Security/Authentication/test/OpenIdConnect/OpenIdConnectTests.cs]: RemoteSignOut

Comment (Tratcher) [reply src/Security/Authentication/test/OpenIdConnect/OpenIdConnectTests.cs]: Add one positive test where the iss and sub are provided and you do get the Set-Cookie header.

Comment (Kahbazi) [reply src/Security/Authentication/test/OpenIdConnect/OpenIdConnectTests.cs]: Did you mean iss and sid?

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**